multimodal/src/autogluon/multimodal/learners/default_learner.py

```diff
@@ -0,0 +1,2906 @@
+        #         temperature=config.distiller.temperature,
+        #         output_feature_loss_weight=output_feature_loss_weight,
+        #         hard_label_loss_func=loss_func,
+        #         soft_label_loss_func=soft_label_loss_func,
+        #         softmax_regression_loss_func=softmax_regression_loss_func,
+        #         output_feature_adaptor=output_feature_adaptor,
+        #         output_feature_loss_func=output_feature_loss_func,
+        #         rkd_loss_func=rkd_loss_func,
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+        # elif self._problem_type == NER:
+        #     task = NerLitModule(
+        #         model=model,
+        #         loss_func=loss_func,
+        #         efficient_finetune=OmegaConf.select(config, "optimization.efficient_finetune"),
+        #         mixup_fn=mixup_fn,
+        #         mixup_off_epoch=OmegaConf.select(config, "data.mixup.turn_off_epoch"),
+        #         model_postprocess_fn=model_postprocess_fn,
+        #         trainable_param_names=trainable_param_names,
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+        # elif self._problem_type == OBJECT_DETECTION:
+        #     task = MMDetLitModule(
+        #         model=model,
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+        # else:
+        #     task = LitModule(
+        #         model=model,
+        #         loss_func=loss_func,
+        #         efficient_finetune=OmegaConf.select(config, "optimization.efficient_finetune"),
+        #         mixup_fn=mixup_fn,
+        #         mixup_off_epoch=OmegaConf.select(config, "data.mixup.turn_off_epoch"),
+        #         model_postprocess_fn=model_postprocess_fn,
+        #         trainable_param_names=trainable_param_names,
+        #         skip_final_val=OmegaConf.select(config, "optimization.skip_final_val", default=False),
+        #         **metrics_kwargs,
+        #         **optimization_kwargs,
+        #     )
+
+        task = self._setup_task_lightning_module(
+            trainable_param_names=trainable_param_names,
+            mixup_fn=mixup_fn,
+            loss_func=loss_func,
+            optimization_kwargs=optimization_kwargs,
+            metrics_kwargs=metrics_kwargs,
+        )
+
+        # 20. Setup call backs
+        checkpoint_callback = self._create_checkpoint_callback(minmax_mode=minmax_mode, save_path=save_path, task=task)
+        early_stopping_callback = self._create_early_stop_callback(
+            validation_metric_name=validation_metric_name, minmax_mode=minmax_mode, task=task
+        )
+        lr_callback = self._create_lr_callback()
+        model_summary = pl.callbacks.ModelSummary(max_depth=1)
+        callbacks = [
+            checkpoint_callback,
+            early_stopping_callback,
+            lr_callback,
+            model_summary,
+        ]
```

Comment:
Consider using a function `_get_lightning_callbacks()` to cover this block. The function can return a list of callbacks.

Reply:
Done